### ISSUE TITLE
autoUpdater event update-downloaded called with null

### ISSUE THREAD

**Issue by safareli** [electron-userland/electron-builder#5659]
* **Version**: 22.9.1
* **Electron Version**: 11.2.2
* **electron-updater Version**: 4.3.5
* **Target**: macOS (darwin 19.6.0),

On macOS I've couple times noticed that autoUpdater event `update-downloaded`  hander was called with `null`. If it's valid then it's not documented.

code looks like this:
```ts
import { autoUpdater, UpdateInfo } from "electron-updater";
....
autoUpdater.on("update-downloaded", (updateInfo: UpdateInfo) => {
  console.log(updateInfo.version)
})
```
and the error like this:
```
TypeError: Cannot read property 'version' of null
    at c.<anonymous> (/Applications/MySuperApp.app/Contents/Resources/app.asar/app/main/dist/ui.prod.js:1:10635)
    at c.emit (events.js:315:20)
    at c.EventEmitter.emit (domain.js:483:12)
    at c.dispatchUpdateDownloaded (/Applications/MySuperApp.app/Contents/Resources/app.asar/app/main/dist/ui.prod.js:17:125992)
    at AutoUpdater.<anonymous> (/Applications/MySuperApp.app/Contents/Resources/app.asar/app/main/dist/ui.prod.js:17:137359)
    at AutoUpdater.emit (events.js:315:20)
    at AutoUpdater.EventEmitter.emit (domain.js:483:12)
```

**Comment by mmaietta** [comment]
Can you please double-check that the file was indeed downloaded? And/or a minimum reproducible repro for me to start with?

I see that it's failing here:
https://github.com/electron-userland/electron-builder/blob/master/packages/electron-updater/src/MacUpdater.ts#L130

And the `event` is defined in the parent `done` closure but TS compiler isn't throwing any errors regarding var scope.
https://github.com/electron-userland/electron-builder/blob/master/packages/electron-updater/src/MacUpdater.ts#L60

**Comment by safareli** [comment]
Unfortunately I cant, a some users has encountered this error couple times, but can't reproduce it, nor have I more details :(
I mainly opened this to document the issue, if others notice similar issue, maybe they could provide more details. This could be closed if it you think it's not actionable.

**Comment by ramaralo** [comment]
I can confirm this as well. It happened today for the first time.
On my case it seems like the event `update-downloaded` fired twice and the second time it was fired with `null`.

Here is a small subset from our logs:
```bash
2021-04-13T08:56:41.148+01:00 [INFO ] [*****] New version downloaded. Auto updater info: { version: '6.15.0-alpha.10',
2021-04-13T08:56:41.172+01:00 [INFO ] [*****] /1618300086462-3180.zip requested
2021-04-13T08:56:41.172+01:00 [INFO ] [*****] /1618300086462-3180.zip requested by Squirrel.Mac, pipe /******-6.15.0-alpha.10-677-mac.zip
2021-04-13T08:56:46.490+01:00 [INFO ] [*****] Proxy server for native Squirrel.Mac is closed (was started to download ******)
2021-04-13T08:56:49.279+01:00 [INFO ] [******] New version downloaded. Auto updater info: null
```

Important part is two log entries for the `update-downloaded` event:
```bash
2021-04-13T08:56:41.148+01:00 [INFO ] [*****] New version downloaded. Auto updater info: { version: '6.15.0-alpha.10',
(...)
2021-04-13T08:56:49.279+01:00 [INFO ] [******] New version downloaded. Auto updater info: null
```

The logs come from this JS:

```javascript
autoUpdater.on(AUTO_UPDATER_EVENTS.UPDATE_DOWNLOADED, (updateInfo: UpdateInfo) => {
        log.info("New version downloaded. Auto updater info:", updateInfo);

        (...)
    });
```

Our versions:
```bash
electron-builder@22.7.0
electron@12.0.0
```

**Comment by ramaralo** [comment]
It hasn't happened again.
We are on `electron-builder` version `22.11.4`

**Comment by ramaralo** [comment]
Update:
It is still happening. It still appears to be random though.

**Comment by mmaietta** [comment]
I found the code, but I don't fully understand what the MacUpdater is doing. It seems a local proxy server isn't receiving a download event payload
https://github.com/electron-userland/electron-builder/search?q=dispatchUpdateDownloaded
https://github.com/electron-userland/electron-builder/blob/71d37044d348ab0cd8f0f9f7754bf6ebaa4b996c/packages/electron-updater/src/MacUpdater.ts#L130-L139

The `event` payload is created here:
https://github.com/electron-userland/electron-builder/blob/71d37044d348ab0cd8f0f9f7754bf6ebaa4b996c/packages/electron-updater/src/AppUpdater.ts#L592-L595

The `updateInfo`:
https://github.com/electron-userland/electron-builder/blob/71d37044d348ab0cd8f0f9f7754bf6ebaa4b996c/packages/electron-updater/src/AppUpdater.ts#L568

What Provider/Publish configs do you all have? I'm curious if there's a pattern

**Comment by ramaralo** [comment]
@mmaietta 

On our case, we're using a `generic` provider. 
Meanwhile while looking into our logs, it's possible that we're getting double notifications for the `update-available` as well and not just `update-download` as I've mentioned [before](update-available).

**Comment by mmaietta** [comment]
Perhaps that could be related to this and potentially resolved?
There was a double event of "update-downloaded" due to the double `checkForUpdates`.
https://github.com/electron-userland/electron-builder/pull/5989

**Comment by ramaralo** [comment]
@mmaietta 
Thanks for taking a look at this.

From the PR I would assume the issue would be constant and reproducible and it seams to be random...
I do see it was published at v22.11.8 and we are art v22.11.7.

I'll try to update and see if it still happens.
